### ISSUE TITLE
Different AUC results in R (catboost.train) and Python (CatBoostClassifier)

### ISSUE THREAD

**Issue by Vikuliia** [catboost/catboost#1126]
Hi! Can there be different AUC results (for test split) in R compare Python in case of same version (20.1), same dataset, random seed, parameters, split and loss function?

**Comment by Avichi** [comment]
What is the error variance in AUC values? It could be the way double is treated in a recurssive calculation within the model?

-----------------------------------
PRIVACY AND CONFIDENTIALITY NOTICE:

This communication constitutes an electronic communication within the meaning of the Electronic Communications Act, 18 USC 2510 and any files transmitted with it may contain PRIVILEGED and CONFIDENTIAL information and may be read or used only by the intended recipient named above, snooping (MITM) while an email is in "real-time" transmission between sender and receiver is a federal crime under the Electronic Communications Protection Act (ECPA), Section 18 U.S.C.A 2517 (4)-Section 18 U.S.C.A 27 Chapter 121 (2701-2712) and Stored Communications Act (SCA, 18 U.S.C. § 2701 and Section 18 U.S.C. § 3121 - U.S. Code - Unannotated Title 18. Crimes and Criminal Procedure § 3121. General prohibition on pen register and trap and trace device use. ) The contents may contain Technical Data that is controlled by the ITAR, 22CFR parts 120 to 130, or EAR, 15CFR parts 730 to 772, and therefore should not be disclosed to a non-U.S. person, or entity, in the U.S. or abroad. If you are not the intended recipient of the email or any of its attachments, please be advised that you have received this email in error and that any use, archiving,physical paper storing,cloud devices storage ,dissemination, distribution, forwarding, printing, reverse engineering, editing and/or copying of this email or any attached files is strictly prohibited. If you have received this email in error or have deliberately(Intentionally) stored this message you will be liable under 18 U.S. Code § 2520, please immediately purge, delete, erase it and all attachments and notify the sender by reply email or contact the sender at the number listed above if one is provided. The sender therefore does not accept liability for any errors or omissions in the contents of this message, which arise as a result of e-mail transmission.All Privacy Protection & Confidentiality apply to all email authors and recipients all American Citizens as per FISA Section 702 and Fourth Amendment of US Constitution. All META Data of this email source file are protected by above applicable laws and cannot be used against the author/receipts in the Court of Law.This email is sent using Microsoft Outlook and Google Mail SMTP/IMAP/POP3 protocols and AndroidOS

________________________________

**Comment by annaveronika** [comment]
Maybe you are using different versions of the library. Or maybe you are using GPU training which is non deterministic.

**Comment by annaveronika** [comment]
Also you might be using numpy or pandas dataframe for python. Try training from file for both libs, the result should be the same.

**Comment by Vikuliia** [comment]
Hi! We have solved the problem. Initially, we had different versions of Catboost for python (20.1) and R (18.1). We updated the version of Catboost in the R studio through a local file (in the list of packages it was clearly visible that version is 20.1). But got different results. Having tried different options, we achieved the same result, updating the catboost directly through R Idle, not R Studio. It’s not clear why there.
